server/job.py

```diff
@@ -1,6 +1,70 @@
 class Job:
+    """
+        The help job being processed by the remote helper.
+        It has three states, described by the tuple
+        (self._complete, self._resolved):
+            started:  (False, False)
+            complete: (True, False)
+            resolved: (True, True)
+
+        It is created in 'started' state, can then be switched
+        to 'complete' state, and is only switched to 'resolved'
+        state when it is queried for its status and it has
+        already been completed.
+    """
     def __init__(self, events_url):
         self._events_url = events_url
+        self._complete = False
+        self._resolved = False
 
-    def get_events_url(self):
+    @property
+    def events_url(self):
         return self._events_url
+
+    @property
+    def done_text(self):
+        return self._done_text
+
+    @done_text.setter
+    def done_text(self, given_done_text):
+        self._done_text = given_done_text
+
+    @property
+    def is_complete(self):
+        return self._complete
+
+    def complete(self):
+        self._complete = True
+
+    def is_running(self):
+        return not self._complete
+
+    def is_resolved(self):
+        return self._resolved
+
+    @property
+    def status(self):
```

Comment:
`status` seems like a bad name to me, but I don't know what name could be better